### PR TITLE
feat: removed error-ex as dependency

### LINKED ISSUE #38: Remove error-ex as dependency
Would you accept a PR that will remove "error-ex" as a dependency?

The current implementation could be switched to 

Example:
```
class JSONError extends Error {
  constructor(message) {
    super(message);
    this.name = 'JSONError';
    this.fileName = undefined
    this.codeFrame = undefined
    this.rawCodeFrame = undefined
  }
}
```

### PR DESCRIPTION
**Pull request by prisis**
This will close #38 

Old error object
```
ErrorEXError [JSONError]: Unexpected token "}" (0x7D) in JSON at position 16 while parsing "{\n\t\"foo\": true,\n}" in foo.json 

  1 | {
  2 |   "foo": true,
> 3 | }
    | ^

    at parseJson (file:///parse-json/index.js:52:20)
    at t.throws.message (file:///parse-json/test.js:56:4)
    at ExecutionContext.throws (file:///parse-json/node_modules/ava/lib/assert.js:487:14)
    at file:///parse-json/test.js:54:4
    at Test.callFn (file:///parse-json/node_modules/ava/lib/test.js:523:21)
    at Test.run (file:///parse-json/node_modules/ava/lib/test.js:536:23)
    at Runner.runSingle (file:///parse-json/node_modules/ava/lib/runner.js:285:33)
    at Runner.runTest (file:///parse-json/node_modules/ava/lib/runner.js:367:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 0) {
  fileName: 'foo.json',
  codeFrame: '  1 | {\n  2 | \t"foo": true,\n> 3 | }\n    | ^',
  rawCodeFrame: '  1 | {\n  2 | \t"foo": true,\n> 3 | }\n    | ^'
}

```

New error object
```
JSONError: Unexpected token "}" (0x7D) in JSON at position 16 while parsing "{\n\t\"foo\": true,\n}" in foo.json

  1 | {
  2 |   "foo": true,
> 3 | }
    | ^

    at parseJson (file:///parse-json/index.js:74:20)
    at t.throws.message (file:///parse-json/test.js:56:4)
    at ExecutionContext.throws (file:///parse-json/node_modules/ava/lib/assert.js:487:14)
    at file:///parse-json/test.js:54:4
    at Test.callFn (file:///parse-json/node_modules/ava/lib/test.js:523:21)
    at Test.run (file:///parse-json/node_modules/ava/lib/test.js:536:23)
    at Runner.runSingle (file:///parse-json/node_modules/ava/lib/runner.js:285:33)
    at Runner.runTest (file:///parse-json/node_modules/ava/lib/runner.js:367:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 0) {
  fileName: 'foo.json',
  codeFrame: '  1 | {\n  2 | \t"foo": true,\n> 3 | }\n    | ^',
  rawCodeFrame: '  1 | {\n  2 | \t"foo": true,\n> 3 | }\n    | ^'
}

```

### CODE REVIEW

**Comment by sindresorhus** [comment]
Have you made sure the error has the same behavior as before?

**Comment by prisis** [comment]
Do you have examples, so I would add it to the tests?

File: index.js [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
-});
+export class JSONError extends Error {
+	constructor(message) {
+		super(message);
+
+		let _messages = message instanceof Error
+			? message.message
+			: message;
+
+		delete this.message;
```
Comment (sindresorhus) [inline index.js:13]: I don't see the point in deleting this.

File: index.js [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
 
-export const JSONError = errorEx('JSONError', {
-	fileName: errorEx.append('in %s'),
-	codeFrame: errorEx.append('\n\n%s\n'),
-});
+export class JSONError extends Error {
+	constructor(message) {
+		super(message);
+
+		let _messages = message instanceof Error
```
Comment (sindresorhus) [inline index.js:9]: Why plural?

File: index.js [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
+			},
+			set(value) {
+				_messages = value;
+			},
+		});
+
+		this.name = 'JSONError';
+		this.fileName = undefined;
+		this.codeFrame = undefined;
+		this.rawCodeFrame = undefined;
```
Comment (sindresorhus) [inline index.js:29]: These could maybe use class fields.

File: index.js [hunk 4]
```diff
@@ -1,12 +1,34 @@
...
+export class JSONError extends Error {
+	constructor(message) {
+		super(message);
+
+		let _messages = message instanceof Error
+			? message.message
+			: message;
+
+		delete this.message;
+
```
Comment (sindresorhus) [inline index.js:14]: You may need:

```js
if (Error.captureStackTrace) {
  Error.captureStackTrace(this, JSONError);
}
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
> Do you have examples, so I would add it to the tests?

I don't. Just making sure you have considered it.

File: index.js (hunk 4 above)
Comment (prisis) [reply index.js:14]: Please, take a look at the [documentation](https://nodejs.org/docs/latest-v8.x/api/errors.html#errors_new_error_message). The most important part:

> The error.stack property will represent the point in the code at which new Error() was called. Stack traces are dependent on V8's stack trace API. Stack traces extend only to either (a) the beginning of synchronous code execution, or (b) the number of frames given by the property Error.stackTraceLimit, whichever is smaller.

.stack will be added automatically.

But to be safe, i will add it

File: index.js (hunk 2 above)
Comment (prisis) [reply index.js:9]: copilot...

**Comment by prisis** [comment]
Did all the testing that is connected to this lib, sure the new Error class is missing the append and line functions like from the old Error.

From my testing it has the same behavior like before, it can have the file name if you run the func and it can be modified after it was thrown.

File: index.js [hunk 5]
```diff
@@ -1,12 +1,37 @@
...
 import {LinesAndColumns} from 'lines-and-columns';
 
-export const JSONError = errorEx('JSONError', {
-	fileName: errorEx.append('in %s'),
-	codeFrame: errorEx.append('\n\n%s\n'),
-});
+export class JSONError extends Error {
+	fileName = undefined;
+	codeFrame = undefined;
+	rawCodeFrame = undefined;
```
Comment (sindresorhus) [inline index.js:8]:
```suggestion
	fileName;
	codeFrame;
	rawCodeFrame;
```

**Pull request merged**